internal/nostr/nostr.go

```diff
@@ -321,78 +320,76 @@ func (svc *Service) NIP47Handler(c echo.Context) error {
 				State: "ALREADY_PUBLISHED",
 			})
 		}
+	} else {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":      requestData.SignedEvent.ID,
+			"walletPubkey": requestData.WalletPubkey,
+			"relayUrl":     requestData.RelayUrl,
+			"webhookUrl":   requestData.WebhookUrl,
+		}).Info("Storing request event")
+	
+		requestEvent = RequestEvent{
+			NostrId: requestData.SignedEvent.ID,
+			Content: requestData.SignedEvent.Content,
+		}
+	
+		err := svc.db.Create(&requestEvent).Error
+		if err != nil {
+			return c.JSON(http.StatusInternalServerError, ErrorResponse{
+				Message: "Failed to store request event",
+				Error:   err.Error(),
+			})
+		}
 	}
 
-	svc.Logger.WithFields(logrus.Fields{
-		"eventId":      requestData.SignedEvent.ID,
-		"walletPubkey": requestData.WalletPubkey,
-		"relayUrl":     requestData.RelayUrl,
-		"webhookUrl":   requestData.WebhookUrl,
-	}).Info("Storing request event and subscription")
-
-	subscription = Subscription{
-		RelayUrl:   requestData.RelayUrl,
-		WebhookUrl: requestData.WebhookUrl,
-		Open:       true,
-		Authors:    &[]string{requestData.WalletPubkey},
-		Kinds:      &[]int{NIP_47_RESPONSE_KIND},
-		Tags:       &nostr.TagMap{"e": []string{requestData.SignedEvent.ID}},
-		Since:      time.Now(),
-		Limit:      1,
-	}
-
-	err := svc.db.Create(&subscription).Error
-	if err != nil {
-		return c.JSON(http.StatusInternalServerError, ErrorResponse{
-			Message: "Failed to store subscription",
-			Error:   err.Error(),
-		})
-	}
-
-	requestEvent = RequestEvent{
-		NostrId: requestData.SignedEvent.ID,
-		Content: requestData.SignedEvent.Content,
-		SubscriptionId: subscription.ID,
-	}
-
-	err = svc.db.Create(&requestEvent).Error
-	if err != nil {
-		return c.JSON(http.StatusInternalServerError, ErrorResponse{
-			Message: "Failed to store request event",
-			Error:   err.Error(),
-		})
+	subscription := Subscription{
+		RelayUrl:     requestData.RelayUrl,
+		WebhookUrl:   requestData.WebhookUrl,
+		Open:         true,
+		Authors:      &[]string{requestData.WalletPubkey},
+		Kinds:        &[]int{NIP_47_RESPONSE_KIND},
+		Tags:         &nostr.TagMap{"e": []string{requestData.SignedEvent.ID}},
+		Since:        time.Now(),
+		Limit:        1,
+		RequestEvent: requestData.SignedEvent,
+		RequestID:    requestEvent.ID,
+		EventChan:    make(chan *nostr.Event, 1),
 	}
 
 	if subscription.WebhookUrl != "" {
-		go func() {
-			ctx, cancel := context.WithTimeout(svc.Ctx, 90*time.Second)
-			defer cancel()
-			event, _, err := svc.processRequest(ctx, &subscription, &requestEvent, &requestData)
-			if err != nil {
-				svc.Logger.WithError(err).Error("Failed to process request for webhook")
-				// what to pass to the webhook?
-				return
-			}
-			svc.postEventToWebhook(event, requestData.WebhookUrl)
-		}()
+		go svc.startSubscription(svc.Ctx, &subscription)
```

Comment:
@bumi mentioned:
> I might think a bit too complicated there and maybe the two options (webhook + sync) is not needed

But aren't they both separate, and good usecases? otherwise you are forced to use a webhook?

Reply:
both are good and important use cases. not sure if we should continue to have it in one endpoint though if it's too complicated.

Reply:
Now that we simplified code, it's actually easier to support both in a single endpoint...